test/unit/sdk/metrics/meter_provider_test.dart

```diff
@@ -0,0 +1,159 @@
+@TestOn('vm')
+
+import 'package:logging/logging.dart';
+import 'package:opentelemetry/sdk.dart' as sdk;
+
+//The following requirements were used to determine what should be tested.
+//[https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#meterprovider]
+
+import 'package:test/test.dart';
+
+void main() {
+  group('MeterProvider:', () {
+    setUp(() {
+      Logger.root.level = Level.ALL; // defaults to Level.INFO
+      Logger.root.onRecord.listen((record) {
+        print('${record.level.name}: ${record.time}: ${record.message}');
+      });
```

Comment:
#fyi there's a really useful [`printOnFailure` utility](https://api.flutter.dev/flutter/test_api.scaffolding/printOnFailure.html) provided by the test package that comes in handy for things like this:

```suggestion
      Logger.root.level = Level.ALL; // defaults to Level.INFO
      Logger.root.onRecord.listen((record) {
        printOnFailure('${record.level.name}: ${record.time}: ${record.message}');
      });
```

This way, you'll only see the logs from a test if it fails without having to always clutter up the test run output.

Also, depending on what you're going for, the default `toString()` representation provided by `LogRecord` looks like so: `'[${level.name}] $loggerName: $message'` - so you might even be able to simplify this to:

```suggestion
      Logger.root.level = Level.ALL; // defaults to Level.INFO
      Logger.root.onRecord.listen(printOnFailure);
      });
```

Reply:
I tried the shortened version but it didn't like the type mismatch so I went with your first suggestion.
![Uploading Screen Shot 2022-09-11 at 8.13.30 AM.png…]()